integrations/sentry/clients/sentry.py

```diff
@@ -23,6 +39,43 @@ def __init__(
         self.client = http_async_client
         self.client.headers.update(self.base_headers)
         self.selector = cast(SentryResourceConfig, event.resource_config).selector
+        self.default_semaphore = default_semaphore
+        self.issues_semaphore = issues_semaphore
+
+    async def fetch_with_rate_limit_handling(
+        self,
+        url: str,
+        params: dict[str, Any] | None = None,
+        semaphore: asyncio.Semaphore | None = None,
+    ) -> httpx.Response:
+        """Rate limit handler
+        This method makes sure requests aren't abusing Sentry's rate-limits.
+        It references:
+        Requests per "window" - By adding a sleep after each request that had a RATE_LIMIT_REMAINING below a threshold
+        for more information about Sentry's rate limits, please check out https://docs.sentry.io/api/ratelimits/
+        """
+        if not semaphore:
+            semaphore = self.default_semaphore
+        while True:
+            async with semaphore:
+                response = await self.client.get(url, params=params)
+            rate_limit_remaining = int(
+                response.headers["X-Sentry-Rate-Limit-Remaining"]
```

Comment:
what about `X-Sentry-Rate-Limit-ConcurrentRemaining`? as you are eventually spawning them concurrently kind of

Reply:
The semaphore makes sure that I'm not able to create more requests then the semaphore limit, so that's handled for me.